### PR TITLE
format db query results in the repository

### PR DESCRIPTION
**Pull request by kzalite**

### CODE REVIEW

File: node-backend/inProgressWorkspaceRepository.js [hunk 1]
```diff
@@ -3,19 +3,29 @@ var logger = require('./logger');
...
+    logger.debug('getting in progress workspace: ' + workspaceId);
     const query = 'SELECT id, owner, state FROM inProgressWorkspace WHERE id = $1';
     db.query(query,
       [workspaceId],
-      callback);
+      function(error, result) {
+        if (error) {
+          callback(error);
+        } else if (!result.rows.length) {
+          callback('In progress workspace with ID ' + workspaceId + ' not found.');
```
[the remark this answers is not in the document]
Comment (chinook25) [reply node-backend/inProgressWorkspaceRepository.js]: That would mean either special casing this (would have to pass next into repo) or changing our error handler. I would put either in their own refactor/commit.

File: node-backend/orderingRepository.js [hunk 2]
```diff
@@ -3,14 +3,22 @@ var logger = require('./logger');
...
-    logger.debug('GET /workspaces/' + workspaceId + '/ordering');
+    logger.debug('getting /workspaces/' + workspaceId + '/ordering');
     db.query('SELECT workspaceId AS "workspaceId", ordering FROM ordering WHERE workspaceId = $1',
       [workspaceId],
-      callback);
+      function(error, result) {
+        if (error) {
+          callback(error);
+        } else if (!result.rows.length) {
+          callback(null, undefined);
```
[the remark this answers is not in the document]
Comment (chinook25) [reply node-backend/orderingRepository.js]: Changed it to callback(null) to keep testing easier :)

File: node-backend/scenarioRepository.js [hunk 3]
```diff
@@ -37,7 +42,15 @@ module.exports = function(db) {
     db.query(
       query,
       [scenarioId],
-      callback
+      function(error, result) {
+        if (error) {
+          callback(error);
+        } else if (!result.rows.length) {
+          callback('No scenario with ID ' + scenarioId + ' found.');
```
[the remark this answers is not in the document]
Comment (chinook25) [reply node-backend/scenarioRepository.js]: See other answer

File: node-backend/workspaceRepository.js [hunk 4]
```diff
@@ -74,7 +84,15 @@ module.exports = function(db) {
     db.query(
       query,
       [workspaceId],
-      callback
+      function(error, result) {
+        if (error) {
+          callback(error);
+        } else if (!result.rows.length) {
+          callback('No workspace with ID ' + workspaceId + ' found.');
```
[the remark this answers is not in the document]
Comment (chinook25) [reply node-backend/workspaceRepository.js]: ^

File: node-backend/inProgressWorkspaceRepository.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielReid) [reply node-backend/inProgressWorkspaceRepository.js]: i think attaching a `statusCode` is sufficient, see line 149 in index.js.

the util.handleError can be changed to do 
`statusCode: error.statusCode || httpStatus.INTERNAL_SERVER_ERROR,`

I agree a bigger refactor would also be good at some point but doing this more correctly now for these cases is a nice test

**Pull request merged**